src/compiler/core.ts

```diff
@@ -92,6 +92,41 @@ module ts {
         return result;
     }
 
+    export function reduce<T>(array: T[], f: (a: T, x: T) => T): T;

```

Comment:
Due to later refactoring after I added this, I am just going to remove both as they are unused.